source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -459,14 +499,27 @@ connections managed by the pool is at least minPoolSize.
 
 Populating the pool MUST NOT block any application threads. For example, it
 could be performed on a background thread or via the use of non-blocking/async
-I/O.
+I/O. Populating the pool MUST NOT be performed unless the pool is "ready".
+
+If an error is encountered while populating a connection, the pool MUST be
+cleared if the `Connection's <#connection>`_ generation is greater than or equal
+to the pool's generation. The error SHOULD be handled like a network error that
+occurred *Before the handshake completed* according to the `Network error when
+reading or writing`_ section in the SDAM specification, if possible in the
+driver's implementation. If not possible, then the error will get handled that
+way by the existing SDAM machinery next time checkOut is called.
```

Comment:
I incorporated this link into the existing language to ensure this error still gets treated as a network error. Let me know what you think.

Reply:
I think we should remove the "network error" wording since this could fail with a command error as well. When it fails with a command error I expect the topology to inspect both generation and topologyVersion (if present in the response).

Reply:
I think I had this section confused with the section describing the error we return from `checkOut`. I removed all the extra stuff I kept about pre-handshake / network error so it looks more like your original suggestion.